network/src/consensus/transactions.rs

```diff
@@ -90,11 +75,11 @@ impl Transactions {
     ) -> Result<(), NetworkError> {
         if let Ok(tx) = Tx::read(&*transaction) {
             let insertion = {
-                let parameters = self.environment.dpc_parameters();
-                let storage = self.environment.storage().read();
-                let consensus = self.environment.consensus_parameters();
+                let parameters = &self.dpc_parameters();
+                let storage = self.storage().read();
+                let consensus = &self.consensus_parameters();
 
-                if !consensus.verify_transaction(parameters, &tx, &storage)? {
+                if !consensus.verify_transaction(&parameters, &tx, &storage)? {
```

Comment:
note: eventually we'll want to have methods like this not require `consensus_parameters` or `storage` to be passed as separate parameters (as they're already available inside `Consensus`; the same applies to the main RPC and miner objects)